src/openforms/submissions/tests/test_submission_attachment.py

```diff
@@ -1617,3 +1617,48 @@ def test_combination_attachment_and_repeating_group_with_numbers(self):
 
         self.assertEqual(len(result), 1)
         self.assertEqual(SubmissionFileAttachment.objects.count(), 1)
+
+    def test_attachment_applies_filename_template(self):
+        components = [
+            {
+                "key": "someFile",
+                "file": {
+                    "name": "prefix_{{ fileName }}_postfix",
+                },
+                "type": "file",
+            }
+        ]
+
+        upload = TemporaryFileUploadFactory.create()
+
+        data = {
+            "someFile": [
+                {
+                    "url": f"http://server/api/v2/submissions/files/{upload.uuid}",
+                    "data": {
+                        "url": f"http://server/api/v2/submissions/files/{upload.uuid}",
+                        "form": "",
+                        "name": upload.file_name,
+                        "size": upload.file_size,
+                        "baseUrl": "http://server",
+                        "project": "",
+                    },
+                    # ignored formio generated data
+                    # "name": "formio generated -guid- filename.foo",
+                    # "size": upload.file_size,
+                    # "type": "image/jpg",
+                    "storage": "url",
+                    "originalName": upload.file_name,
+                }
+            ],
+        }
+
+        submission = SubmissionFactory.from_components(components, data)
+
+        result = attach_uploads_to_submission_step(submission.steps[0])
+        self.assertEqual(len(result), 1)
+
+        attachment_basename = Path(result[0][0].file_name).stem
+        upload_basename = Path(upload.file_name).stem
+
+        self.assertEqual(attachment_basename, f"prefix_{upload_basename}_postfix")
```

Comment:
Can you please instead use literals in the test and factory call so that we can make plain assertions instead of evaluating python code in our head?

I don't really want to have to reason about the datastructure returned by `attach_uploads_to_submission_step` in a test that deals with the filename templating :grimacing: